presto-hive/src/main/java/io/prestosql/plugin/hive/HiveSplitManager.java

```diff
@@ -216,6 +217,8 @@ public ConnectorSplitSource getSplits(
 
         Iterable<HivePartitionMetadata> hivePartitions = getPartitionMetadata(session, metastore, table, tableName, partitions, bucketHandle.map(HiveBucketHandle::toTableBucketProperty));
 
+        // Only one thread per partition is usable when a table is not transactional
+        int concurrency = AcidUtils.isTransactionalTable(table.getParameters()) ? splitLoaderConcurrency : min(splitLoaderConcurrency, partitions.size());
```

Comment:
You could static import `isTransactionalTable`

Reply:
Done.